mountpoint-s3-fs/src/metrics.rs

```diff
@@ -90,12 +93,28 @@ fn poll_process_metrics(sys: &mut System) {
 #[derive(Debug)]
 struct MetricsSink {
     metrics: DashMap<Key, Metric>,
+    otlp_exporter: Option<OtlpMetricsExporter>, // optional OTLP exporter
 }
 
 impl MetricsSink {
-    fn new() -> Self {
+    fn new(otlp_config: Option<OtlpConfig>) -> Self {
+        // Initialise the OTLP exporter if a config is provided
+        let otlp_exporter = otlp_config
```

Comment:
We should probably handle the None otlp_config separately with it's dedicated tracing. This would provide more clarity and possibly avoid confusing with error cases